beacon_node/beacon_chain/src/data_availability_checker/error.rs

```diff
@@ -49,6 +47,8 @@ impl Error {
             | Error::SlotClockError => ErrorCategory::Internal,
             Error::InvalidBlobs { .. }
             | Error::InvalidColumn { .. }
+            | Error::MissingBlobs
+            | Error::MissingCustodyColumns
```

Comment:
The category change makes sense but it doesn't look like we use this to determine peer penalty.

Related issue:
https://github.com/sigp/lighthouse/issues/4546

I think we'll need to handle `BlockError::AvailabilityCheck` in `handle_failed_chain_segment`:
https://github.com/sigp/lighthouse/blob/ee7fca3ebd277a58739c7a19046608b7556afc59/beacon_node/network/src/network_beacon_processor/sync_methods.rs#L711